client/reducers/checks.js

```diff
@@ -1,12 +1,33 @@
 import { PUT_CHECK, DELETE_CHECK } from '../actions/checks'
+import { PENDING, SUCCESS, ERROR } from '../actions/status'
 
-export default function checks(state = [], action) {
-  switch (action.type) {
-    case PUT_CHECK:
-      return state.concat(action.payload)
-    case DELETE_CHECK:
-      break
-    default:
-      return state.filter(check => check.type !== action.payload.type)
+export function init(repo) {
+  return repo.checks.map(c => ({
+    type: c.type,
+    error: false,
+    isUpdating: false,
+    enabled: true
+  }))
+  .reduce((state, check) => {
+    state[check.type] = check
+    return state
+  }, {})
+}
+
+export default function checks(state = {}, action) {
+  const checks = Object.assign({}, state)
+  const checkType = action.payload.type
+  checks[checkType] = checks[checkType] || {}
+  if (action.status === PENDING) {
+    checks[checkType].isUpdating = true
+    checks[checkType].error = false
+  } else if (action.status === SUCCESS) {
+    checks[checkType].isUpdating = false
+    checks[checkType].error = false
+    checks[checkType].enabled = action.type === PUT_CHECK
+  } else if (action.status === ERROR) {
+    checks[checkType].isUpdating = false
+    checks[checkType].error = action.payload
   }
+  return checks
```

Comment:
From my understanding, the [more idiomatic way](http://redux.js.org/docs/basics/Reducers.html) for Redux is to use switch/case. Please also see my other comment about making `checks` a sub-reducer of `repo`.